### PR TITLE
EVG-3026: reduce interface surface for util.HttpClientPools

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: util/http_client_pool.go [hunk 1]
```diff
@@ -13,57 +13,124 @@ import (
...
+	p.pool[token].Put(c)
+}
 
 const httpClientTimeout = 5 * time.Minute
 
+var httpClientPool *sync.Pool
+var oauthClientPool *oauthPool
+
+func newConfiguredBaseTransport() *http.Transport {
+	return &http.Transport{
```
[the remark this answers is not in the document]
Comment (tychoish) [reply util/http_client_pool.go]: This should be *totally* equivalent to the current implementation, except it constructs the same base transport (as we currently do,) and allows us to "reset" in the case where we somehow get handed an HTTP client with a transport that we're not expecting (on line 113).

This change (I think) has us reusing transports slightly more effectively than before, but that's not my core goal here, and I'm also not super inclined to want to do more than clean up the interfaces right now.

File: util/http_client_pool.go [hunk 2]
```diff
@@ -13,57 +13,124 @@ import (
 	"golang.org/x/oauth2"
 )
 
-var httpClientPool *sync.Pool
+type oauthPool struct {
+	mu   sync.RWMutex
+	pool map[string]*sync.Pool
```
[the remark this answers is not in the document]
Comment (tychoish) [reply util/http_client_pool.go]: The map here is of token to pools of clients configured with that token. 

I don't want to move to using shared clients at this point.

File: migrations/zero_date_fix.go [hunk 3]
```diff
@@ -147,11 +147,11 @@ func makeZeroDateMigration(database, githubToken string) db.MigrationOperation {
 
 func githubFetchRealCreateTime(token, owner, repo, revision string) (*time.Time, error) {
 	ctx := context.Background()
-	httpClient, err := util.GetHttpClientForOauth2(token)
```
[the remark this answers is not in the document]
Comment (tychoish) [reply migrations/zero_date_fix.go]: done!

File: util/http_client_pool.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardsamuels) [reply util/http_client_pool.go]: Fair enough, can you throw a ticket on the backlog to actually do this at some point in the future?

**Pull request merged**